hack/update-codegen.sh

```diff
@@ -40,7 +40,7 @@ echo ">> Using ${CODEGEN_PKG} for the code generator"
 chmod +x "${CODEGEN_PKG}/generate-groups.sh"
 chmod +x "${CODEGEN_PKG}/generate-internal-groups.sh"
 
-PROJECT_ROOT=${GOPATH}/src/github.com/kubeflow/katib
+PROJECT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
```

Comment:
Why do we need to change it ?
I guess, user needs to git clone repo to the `${GOPATH}/src/github.com/kubeflow/katib` for Go packages.

Reply:
IIUC, we need to put this configuration here. That is the limitation of buf.